src/game/TileEngine/Explosion_Control.cc

```diff
@@ -1168,11 +1168,33 @@ static BOOLEAN ExpAffect(const INT16 sBombGridNo, const INT16 sGridNo, const UIN
 	 }
 	 else
 	 {
-			SOLDIERTYPE* const pSoldier = WhoIsThere2(sGridNo, bLevel);
-			if (pSoldier == NULL) return fRecompileMovementCosts;
-			// someone is here, and they're gonna get hurt
-
-			fRecompileMovementCosts = DishOutGasDamage(pSoldier, pExplosive, sSubsequent, fRecompileMovementCosts, sWoundAmt, sBreathAmt, owner);
+		 SOLDIERTYPE* const pSoldier = WhoIsThere2(sGridNo, bLevel);
+		 if (pSoldier == NULL) return fRecompileMovementCosts;
+		 // someone is here, and they're gonna get hurt
+
+		 EXPLOSIVETYPE const* pExplosive = 0;
+		 INT8 soldierbLevel = gpWorldLevelData[pSoldier->sGridNo].ubExtFlags[pSoldier->bLevel];
+		 // Look for gas
+		 if (!AM_A_ROBOT(pSoldier))
+		 {
+			 if (soldierbLevel & MAPELEMENT_EXT_TEARGAS)
+			 {
+				 pExplosive = &Explosive[GCM->getItem(TEARGAS_GRENADE)->getClassIndex()];
+			 }
+			 if (soldierbLevel & MAPELEMENT_EXT_MUSTARDGAS)
+			 {
+				 pExplosive = &Explosive[GCM->getItem(MUSTARD_GRENADE)->getClassIndex()];
```

Comment:
Sorry, realised something else. If the two gases are both present, the latter can override pExplosive. So I guess it would be better to use |= for assignment?

Reply:
As far as I tested there can't be 2 or more gases on one spot. There is kind of a priority for the gases leading to normal smoke getting overwritten by mustard gas which get's overwritten by creatures gas.

Here a screenshot in which i threw a tear gas grenade right on a mustard gas grenade
![1](https://user-images.githubusercontent.com/9992455/31718000-80d935d8-b40e-11e7-844f-eb48277e0e14.png)

Reply:
Okay, wrong. There can be 2 or more gases present but only 1 is visible at a time.

Reply:
Okay, wrong again. There can be 2 gases on one spot but only one gas is affecting mercs at a time you can't stack grenade effects.
I saw mustard gas decreasing BP and thought the reason for this the underlying tear gas. But mustard gas is decreasing BP and HP.

Reply:
So how do you know that both are not being applied?